File: Lab41/app/src/main/java/com/example/lab41/MainActivity.java
package com.example.lab41;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity implements OnOptionChecked{
    DynamicFragment dynamicFragment = new DynamicFragment();
    SecondDynamicFragment secondDynamicFragment = new SecondDynamicFragment();
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    public void onOptionChecked(int id) {
        Fragment f;
        if (id == R.id.showF1){
            f  = dynamicFragment;
        }else {
            f = secondDynamicFragment;
        }
        FragmentManager fm = getSupportFragmentManager();
        FragmentTransaction ft = fm.beginTransaction();

        ft.replace(R.id.fragmentActionViewer, f);
        ft.commit();
    }
}

File: Task2/app/src/main/java/com/example/task2/MainActivity.kt
package com.example.task2

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
    fun openSimple(view : View){
        val navigateToSimpleList = Intent(this, SimpleList::class.java)
        startActivity(navigateToSimpleList)
    }
    fun openMultiple(view : View){
        val navigateToMultiChoiceList = Intent(this, MultiChoiceList::class.java)
        startActivityForResult(navigateToMultiChoiceList, 0)
    }
    fun openGrid(view : View){
        val navigateToGridList = Intent(this, GridList::class.java)
        navigateToGridList.putExtra("ARG1", "String")
        navigateToGridList.putExtra("ARG2", 10)
        navigateToGridList.putExtra("ARG3", false)
        startActivity(navigateToGridList)
    }
    fun openCustom(view : View){
        val navigateToCustomList = Intent(this, CustomList::class.java)
        startActivity(navigateToCustomList)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (resultCode == 0)
            Toast.makeText(this, "Nothing selected", Toast.LENGTH_SHORT).show()
        else
            if (data != null) {
                Toast.makeText(this, data.getStringExtra("Result"), Toast.LENGTH_SHORT).show()
            }
    }
}